web/src/timerender.ts

```diff
@@ -355,6 +387,28 @@ export function get_markdown_time_tooltip(reference: HTMLElement): DocumentFragm
     return "";
 }
 
+function update_recent_topics_timestamps(): void {
+    if (!$('[id*="recent_conversation"]').is(":visible")) {
+        // This condition serves as an alternative to 'recent_topics_util.is_visible()'
+        // because we cannot import 'recent_topics_util' until it is converted to TypeScript.
+        // Once the conversion is done, we should use the 'is_visible' method from there.
+        return;
+    }
```

Comment:
The file is tiny, it should take hardly any time to convert to typescript; we should just do that.

```
$ cat web/src/recent_view_util.js 
let is_rt_visible = false;

export function set_visible(value) {
    is_rt_visible = value;
}

export function is_visible() {
    return is_rt_visible;
}

export function get_topic_key(stream_id, topic) {
    return stream_id + ":" + topic.toLowerCase();
}

export function get_key_from_message(msg) {
    if (msg.type === "private") {
        // The to_user_ids field on a direct message object is a
        // string containing the user IDs involved in the message in
        // sorted order.
        return msg.to_user_ids;
    } else if (msg.type === "stream") {
        return get_topic_key(msg.stream_id, msg.topic);
    }
    throw new Error(`Invalid message type ${msg.type}`);
}
```

Reply:
Yeah, that makes sense. I added a prep commit to convert the module to Typescript.